test/server/cards/01_SOR/units/BlackOneScourgeOfStarkillerBase.spec.ts

```diff
@@ -0,0 +1,67 @@
+                    phase: 'action',
+                    player1: {
+                        hand: ['black-one#scourge-of-starkiller-base'],
+                        deck: ['confiscate', 'waylay', 'isb-agent', 'cartel-spacer', 'wampa', 'disarm', 'atst']
+                    },
+                    player2: {
+                        hand: ['vanquish', 'rivals-fall']
+                    }
+                });
+            });
+
+            it('draw 3 cards if hand is discarded', function () {
+                const { context } = contextRef;
+
+                // Player 1 plays Black One and activates When Played, drawing cards
+                context.player1.clickCard(context.blackOne);
+
+                expect(context.player1.handSize).toBe(0);
+                expect(context.player1).toHaveEnabledPromptButton('You may discard your hand');
+                context.player1.clickPrompt('You may discard your hand');
+
+                expect(context.player1.handSize).toBe(3);
+                expect(context.confiscate).toBeInLocation('hand', context.player1);
+                expect(context.waylay).toBeInLocation('hand', context.player1);
+                expect(context.isbAgent).toBeInLocation('hand', context.player1);
+
+                // Player 2 plays Vanquish and defeats Black One
+                expect(context.player2).toBeActivePlayer();
+                context.player2.clickCard(context.vanquish);
+
+                // Player 1 actives When Defeated, drawing cards
+                expect(context.player1).toHaveEnabledPromptButton('You may discard your hand');
+                context.player1.clickPrompt('You may discard your hand');
+
+                expect(context.player1.handSize).toBe(3);
+                expect(context.cartelSpacer).toBeInLocation('hand', context.player1);
+                expect(context.wampa).toBeInLocation('hand', context.player1);
+                expect(context.disarm).toBeInLocation('hand', context.player1);
+
+                // Player 1 plays Black One again and activates When Played, taking damage
+                expect(context.player1).toBeActivePlayer();
+                context.player1.moveCard(context.blackOne, 'hand');
+                context.player1.clickCard(context.blackOne);
+                context.player1.clickPrompt('You may discard your hand');
+
+                expect(context.player1.handSize).toBe(1);
+                expect(context.atst).toBeInLocation('hand', context.player1);
+                expect(context.p1Base.damage).toBe(6);
+
+                // Player 2 plays Rival's Fall and defeats Black One
+                expect(context.player2).toBeActivePlayer();
+                context.player2.clickCard(context.rivalsFall);
+
+                // Player 1 actives When Defeated, drawing cards
+                context.player1.clickPrompt('You may discard your hand');
+
+                expect(context.player1.handSize).toBe(0);
+                expect(context.p1Base.damage).toBe(15);
```

Comment:
This last test feels redundant since we've already confirmed that draw damage happens, unless there's something I'm missing

Reply:
Yeah, it's a bit redundant but the other one was testing the "when played" ability, this is the "when defeated" one. I would keep both to be safe.